From m.kozlowski@tuxland.pl  Wed Mar 11 14:35:44 2009
From: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Date: Tue,  3 Mar 2009 19:38:35 +0100
Subject: Staging: comedi: usbdux: don't initialize semaphores on init
To: gregkh@suse.de
Cc: m.kozlowski@tuxland.pl
Message-ID: <1236105518-19565-6-git-send-email-m.kozlowski@tuxland.pl>


This patch removes usbdux semaphores initialization
from init function as they are initialized later on in
probe function. Also remove init_usb_devices() as it is
not needed anymore.

Signed-off-by: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Cc: Bernd Porr <BerndPorr@f2s.com>
Cc: Ian Abbott <abbotti@mev.co.uk>
Cc: Frank Mori Hess <fmhess@users.sourceforge.net>
Cc: David Schleef <ds@schleef.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/staging/comedi/drivers/usbdux.c |   12 ------------
 1 file changed, 12 deletions(-)

--- a/drivers/staging/comedi/drivers/usbdux.c
+++ b/drivers/staging/comedi/drivers/usbdux.c
@@ -2910,17 +2910,6 @@ static comedi_driver driver_usbdux = {
       .detach =		usbdux_detach,
 };
 
-static void init_usb_devices(void)
-{
-	int index;
-
-	/* all devices entries are invalid to begin with */
-	/* they will become valid by the probe function */
-	/* and then finally by the attach-function */
-	for (index = 0; index < NUMUSBDUX; index++)
-		init_MUTEX(&(usbduxsub[index].sem));
-}
-
 /* Table with the USB-devices: just now only testing IDs */
 static struct usb_device_id usbduxsub_table[] = {
 	{USB_DEVICE(0x13d8, 0x0001) },
@@ -2945,7 +2934,6 @@ static int init_usbdux(void)
 {
 	printk(KERN_INFO KBUILD_MODNAME ": "
 	       DRIVER_VERSION ":" DRIVER_DESC "\n");
-	init_usb_devices();
 	usb_register(&usbduxsub_driver);
 	comedi_driver_register(&driver_usbdux);
 	return 0;
